misc/l10n_testing.md

```diff
@@ -14,44 +14,110 @@ Each Mozilla project has its own process for taking your translations and making
+
+![UI layout bug](/assets/images/l10n_errors/UI_bug.png)
+
 ### Mistranslation
 
+Mistranslation occurs when meaning is lost between the source language and the target language. This can be caused accidentally by a lack of context for where a translation will be placed within a user interface (very common in software localization), a translator's lack of subject matter expertise, or intentionally due to user interface text constraints.
+
+![Mistranslation bug](/assets/images/l10n_errors/mistranslation.png)
+
 ### Untranslated strings
 
+Untranslated strings appear in a user interface when developers don't include that string in the string resource files available for localization, but instead in their code. It's a practice called "hardcoding." The best way to correct this error is to make sure that Pontoon does not contain the string for translation (or that it's not in the string resource files) and then contact developers using Bugzilla to make the string available for localization.
+
+![Untranslated string bug](/assets/images/l10n_errors/untranslated.png)
+
 ### Style bugs
 
+Mozilla's official style guide serves as a primer to help communities create their own style guides. Style errors are when the tone or appearance of a translation does not match the tone or appearance described in a style guide. These can be corrected by making sure that localizers understand the style guide rules and follow them when translating.
+
+![Style bug](/assets/images/l10n_errors/style_bug.png)
+
 ### Number, date, currency
 
+Every language and region displays numbers, dates, and currency differently. These errors occur when the user sees an incorrect number format, date format, or currency for the language and region of their localized Firefox. This can be fixed by filing a bug in Bugzilla.
+
+![Number bug](/assets/images/l10n_errors/number_bug.png)
+
 ### XML parsing errors
 
+XML parsing errors (aka "the yellow screen of death") cause Firefox to display a yellow error page. This is often due to a broken translated string in that localization of Firefox which causes the Firefox build to break. With l20n, these errors will no longer be relevant.
+
+![XML parsing bug](/assets/images/l10n_errors/xml_parsing.png)
+
 ### Exposed variables and placeholders
 
+Variables and placeholders become exposed in the user interface when the translated string has a typo within the variable or placeholder. Because it is not exactly the variable that the software or website expected to find, the system simply displays the variable rather than its value. A very common example of this is with the variable &browserShortName. If that variable is typed correctly in the translation, it will return the variable's value (in this case, it's `Firefox`). If it contains a typo, the user will see the variable instead (e.g., ``&browsershortname``). These are discoverable in the Mozilla localization dashboards and can be corrected in Pontoon.
```

Comment:
Sorry, confusing comment from me. I meant that there's another brandShortName before that.

This should look like this

```
Variables and placeholders become exposed in the user interface when the translated string has a typo within the variable or placeholder. Because it is not exactly the variable that the software or website expected to find, the system simply displays the variable rather than its value. A very common example of this is with the variable `&browserShortName`. If that variable is typed correctly in the translation, it will return the variable's value (in this case, it's `Firefox`). If it contains a typo, the user will see the variable instead (e.g., `&browsershortname`). These are discoverable in the Mozilla localization dashboards and can be corrected in Pontoon.
```